tensorflow/core/kernels/resize_nearest_neighbor_op.cc

```diff
@@ -152,21 +158,30 @@ struct ResizeNearestNeighbor<CPUDevice, T, half_pixel_centers, align_corners> {
         if (half_pixel_centers) {
           in_y = std::max(static_cast<Eigen::Index>(0), in_y);
         }
-        for (Eigen::Index x = 0; x < out_width; ++x) {
-          Eigen::Index in_x = std::min(
-              (align_corners)
-                  ? static_cast<Eigen::Index>(roundf(scaler(x, width_scale)))
-                  : static_cast<Eigen::Index>(floorf(scaler(x, width_scale))),
-              in_width - 1);
-          if (half_pixel_centers) {
-            in_x = std::max(static_cast<Eigen::Index>(0), in_x);
-          }
-          std::copy_n(&input(b, in_y, in_x, 0), channels, &output(b, y, x, 0));
+        Eigen::Index in_x = std::min(
+            (align_corners)
+                ? static_cast<Eigen::Index>(roundf(scaler(x, width_scale)))
+                : static_cast<Eigen::Index>(floorf(scaler(x, width_scale))),
+            in_width - 1);
+        if (half_pixel_centers) {
+          in_x = std::max(static_cast<Eigen::Index>(0), in_x);
         }
+        std::copy_n(&input(bs, in_y, in_x, 0), channels, &output(bs, y, x, 0));
       }
-    }
+    };
+    Eigen::Index N = batch_size * out_height * out_width;
+    Shard(worker_threads.num_threads, worker_threads.workers, N, 1000.0,
```

Comment:
Let's use `d.ParallelFor` instead of `Shard`. This way we don't have to pass the `OpKernelContext* c` as an extra parameter.

See the code below for example on how to estimate the cost for ParallelFor:
https://github.com/tensorflow/tensorflow/blob/6c092464ff25444afa1c0608cf2542519cb8bbd8/tensorflow/core/kernels/training_ops.cc#L401-L418

(You can multiply each cost term with a constant variable to fine-tune.)

Reply:
use d.ParallelFor instead of Shard.
But can you give more hints about how to estimate the cost? @penpornk  Thanks in advance.

Reply:
After refer to https://eigen.tuxfamily.org/dox/unsupported/TensorCostModel_8h_source.html
I add the estimation of the cost for ParallelFor.